src/components/src/main/java/org/apache/jmeter/visualizers/ViewResultsFullVisualizer.java

```diff
@@ -150,6 +150,7 @@ public void add(final SampleResult sample) {
         synchronized (buffer) {
             buffer.add(sample);
             dataChanged = true;
+            CorrelationRecorder.setBuffer(buffer);
```

Comment:
This does not look correct in terms of isolation of objects. Why is this call needed ?

Reply:
We removed that call and created a listener called Correlation response recorder instead to record the response.

Reply:
We understand it isn't correct. We have removed that illegal call and introduced a new response listener.
The new listener called "Correlation Response recorder" will be responsible to record the API response which is required for Auto-Correlation.
Also, the documentation has been added for the newly added listener.